src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java

```diff
@@ -111,18 +136,21 @@ public static String httpRequest(URI uri, int connectionTimeout, HttpMethod meth
 
         if (headers != null && !headers.isEmpty()) {
             for (Map.Entry<String, String> e : headers.entrySet()) {
-                httpRequestBuilder.setHeader(e.getKey(), e.getValue());
+                httpUriRequest.setHeader(e.getKey(), e.getValue());
             }
         }
 
-        HttpRequest request = httpRequestBuilder.build();
+        CloseableHttpClient httpClient = allowRedirects
+            ? HttpClientBuilder.create().build()
+            : HttpClientBuilder.create().disableRedirectHandling().build();
 
-        try {
-            return client.send(request, java.net.http.HttpResponse.BodyHandlers.ofString());
-        } catch (InterruptedException | IOException e) {
+        try  {
+            return httpClient.execute(httpUriRequest, new HttpResponseHandler());
```

Comment:
Is it possible to pass the `httpUriRequest` to the `HttpResponseHandler` constructor here? That way we could extract the URI directly within `HttpResponseHandler` (to keep from doing it in OtpDispatcherResponse) and that design may help with extracting other information down the road (if needed).

Reply:
@landonreed that is not possible. The `httpUriRequest` is required by `httpClient.execute`. Also the connection is closed after this point so any attempts to down the response body would fail.